runtime/src/transaction/tree.rs

```diff
@@ -17,13 +17,17 @@ use crate::{
 // NOTE: This should be kept in sync with go/runtime/transaction/transaction.go.
 
 #[derive(Debug)]
+#[repr(u8)]
 enum ArtifactKind {
-    Input,
-    Output,
+    _Invalid = 0,
```

Comment:
Why is this needed? Just remove this invalid kind and you will not be able to construct one accidentally?